src/core/lib/surface/server.cc

```diff
@@ -1166,6 +1166,7 @@ void grpc_server_setup_transport(
   chand->server = s;
   server_ref(s);
   chand->channel = channel;
+  chand->socket_node.release();
```

Comment:
Similar issue here.  It would probably be cleaner to give `channel_data` its own ctor and then change `init_channel_elem()` to use placement-new to initialize it.

Reply:
Done.